tf_agents/trajectories/time_step.py

```diff
@@ -30,14 +30,21 @@
 from tf_agents.typing import types
 
 
-def _as_float32_array(a):
-  r = np.asarray(a, dtype=np.float32)
+def _as_array(a, t=np.float32):
+  if t == None:
+    t = np.float32
+  r = np.asarray(a, dtype=t)
   if np.isnan(np.sum(r)):
     raise ValueError('Received a time_step input that converted to a nan array.'
                      ' Did you accidentally set some input value to None?.\n'
                      'Got:\n{}'.format(a))
   return r
 
+def _get_numpy_dt(spec):
+  if type(spec) == tensor_spec.TensorSpec:
```

Comment:
dtype = spec.dtype
if isinstance(dtype, tf.dtypes.DType):
  dtype = dtype.as_numpy_dtype
return np.dtype(dtype)

Reply:
Thank for the suggestion, I had to extend it a bit - we might call `_get_dumpy_dtype` on `None` (e.g. in [testNoneValuesCaught](https://github.com/tensorflow/agents/blob/ad1b4b0d38779c6b36c4387b0479f8887e5e1e45/tf_agents/trajectories/time_step_test.py#L403-L407)) or `float` (e.g. in [testTerminationIsLast](https://github.com/tensorflow/agents/blob/ad1b4b0d38779c6b36c4387b0479f8887e5e1e45/tf_agents/trajectories/time_step_test.py#L83-L87)), so I included a quick exit condition if `spec` is neither `TensorSpec` nor `ArraySpec`